VCDA-3752, 3753, 3855, 3871 : CRS for TKGm v1.5.4 and v1.4.2

arunmk opened the pull request:
This brings in CRS templates for 1.5.4 and 1.4.2. The antrea yaml needs to be installed on the parent cluster (KinD) and the other yaml is to be installed on the child cluster itself. After some time, a `tanzu package installed list -A` will show the packages installed.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182)
<!-- Reviewable:end -->

ltimothy7 reviewed (commented):
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182)* status: all files reviewed, 3 unresolved discussions (waiting on @arunmk and @sahithi)

___
*[`tanzu/crs/v1.4.2/v1.4.2-antrea.yaml.template` line 2097 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182#-N5C3cO5CvRH5Iljoq8f:-N5C3cO5CvRH5Iljoq8g:b587ka2) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/f69018bd4834b52c82d5d8d3ea9d95f3475be924/tanzu/crs/v1.4.2/v1.4.2-antrea.yaml.template#L2097)):*
> ```
>     #@overlay/match-child-defaults missing_ok=True
>     ---
>     infraProvider: vsphere
> ```

Double checking if `vsphere` should be here
___
*[`tanzu/crs/v1.4.2/v1.4.2.yaml.template` line 1600 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182#-N5C3iKc7oBk8lI9Ly7f:-N5C3iKc7oBk8lI9Ly7g:b587ka2) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/f69018bd4834b52c82d5d8d3ea9d95f3475be924/tanzu/crs/v1.4.2/v1.4.2.yaml.template#L1600)):*
> ```
>           tkgVersion: v1.4.2
>           infrastructure:
>             provider: vsphere
> ```

Double checking if `vsphere` should be here
___
*[`tanzu/crs/v1.5.4/v1.5.4-antrea.yaml.template` line 4711 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182#-N5C3Y8k0mGMIkSsQQnM:-N5C3Y8k0mGMIkSsQQnN:b75dxk6) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/f69018bd4834b52c82d5d8d3ea9d95f3475be924/tanzu/crs/v1.5.4/v1.5.4-antrea.yaml.template#L4711)):*
> ```
>     #@overlay/match-child-defaults missing_ok=True
>     ---
>     infraProvider: vsphere
> ```

Should this say a name specific to vcd?


<!-- Sent from Reviewable.io -->

sahithi reviewed (approved):
Couple of questions and comments Arun,

1. What is the main difference between antrea 1.5.4 and 1.4.2? Is there a antrea version change? 
2. I also did not understand why Kapp controller yaml needs to be applied on the target cluster but not KIND VM. kappcontroller is also a CRS definition. CRS is not even enabled on the target cluster, am I correct?
3. Are we not doing metrics server for this release?
4. Can you rename the v1.5.4.yaml.template to v1.5.4-kappcontroller.yaml.template and the same for 1.4.2?
5. . For future reference, where do we need to get these manifest files for different TKG versions? Maybe, a confluence document with the steps will enable others to do this exercise in the future.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> after addressing the above comments

> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/182#-:-N5D2DqGApFwHaCNzV1n:bdmfx1e)* status: all files reviewed, 3 unresolved discussions (waiting on @arunmk)



<!-- Sent from Reviewable.io -->

arunmk commented:
@sahithi thanks for asking these questions. I should have provided the context. The answers to these questions will help.

1. What is the main difference between antrea 1.5.4 and 1.4.2? Is there a antrea version change?
[AK]: Yes, that is one change. The way I get this file is: to get the whole 1.4.2 yaml and split it into two parts. So there may be other differences.
Why I split it into two parts is: antrea is needed for the CPI to function. Hence the KinD VM needs to wait for it to come up. However the other components only need to be installed into the target cluster, and the KinD VM does not need to wait for them to complete starting up. Hence the split. We make the target cluster self-managing (required), move the objects, and then install all of the packages in 1.5.4 (or 1.4.2) and exit the KinD VM.

2. I also did not understand why Kapp controller yaml needs to be applied on the target cluster but not KIND VM. kappcontroller is also a CRS definition. CRS is not even enabled on the target cluster, am I correct?
[AK]: CRS will be enabled on the Target cluster. There are many components that need to be started on the target VM. These packages are in a reconciliatory loop and they can take 10-20 minutes to come up. There is no value in making the KinD VM wait for them to come up.

3. Are we not doing metrics server for this release?
[AK]: The 1.5.2.yaml and the 1.4.2.yaml have metrics-server, kapp-controller, tanzu-addons-manager, tanzu-featuregates (1.5.2 only), tanzu-kubernetes-release, secretgen-controller (1.5.2 only) and a few other packages.

4. Can you rename the v1.5.4.yaml.template to v1.5.4-kappcontroller.yaml.template and the same for 1.4.2?
[AK]: this is all of the packages mentioned in 3 and more. This is essentially all of tanzu packages and secrets except Antrea due to the reason given in (1).


For future reference, where do we need to get these manifest files for different TKG versions? Maybe, a confluence document with the steps will enable others to do this exercise in the future.
[AK]: I was hoping to crowd-source this and let the team do one each. I will make a confluence doc for it.

arunmk commented:
@ltimothy7 thanks for pointing those out. I was not sure what to do there. Do you already know if there is a good replacement for it?

If not, is it okay with you if I take a ticket and check what we can do in the future for it?
